retis/src/core/probe/kernel/probe_stack.rs

```diff
@@ -107,6 +129,10 @@ impl ProbeStack {
                     _ => return Ok(()),
                 };
 
+                mgr.global_options()
+                    .iter()
+                    .try_for_each(|o| probe.set_option(o.clone()))?;
```

Comment:
I understand what you meant in the discussion now. I would prefer this to be handled in the runtime manager before attaching probes as this is already the behavior when using the `ProbeBuilderManager` and this would make the behavior different based on its state. I actually see the lack of this as a bug in my original manager implementation. Also we might even be able to consolidate things and use the same logic (once moved to the runtime manager, and not while converting the manager type).

If for some reason we really want a probe not to follow a global setting, we should add an API for that instead (when the need arises).